chain-api/src/types/launchpadDtos.ts

```diff
@@ -0,0 +1,198 @@
+ * limitations under the License.
+ */
+import BigNumber from "bignumber.js";
+import { IsNotEmpty, IsNumber, IsOptional, IsString, Max, Min } from "class-validator";
+
+import { BigNumberProperty } from "../validators";
+import { IsNonZeroBigNumber } from "../validators";
+import { TokenBalance } from "./TokenBalance";
+import { ChainCallDTO } from "./dtos";
+import { Type } from "class-transformer";
+
+export class CreateTokenSaleDTO extends ChainCallDTO {
+  @IsString()
+  @IsNotEmpty()
+  public tokenName: string;
+
+  @IsString()
+  @IsNotEmpty()
+  public tokenSymbol: string;
+
+  @IsString()
+  @IsNotEmpty()
+  public tokenDescription: string;
+
+  @IsString()
+  @IsNotEmpty()
+  public tokenImage: string;
+
+  @BigNumberProperty()
+  @IsNotEmpty()
+  public preBuyAmount: BigNumber;
+
+  @IsString()
+  @IsOptional()
+  public websiteUrl?: string;
+
+  @IsString()
+  @IsOptional()
+  public telegramUrl?: string;
+
+  @IsString()
+  @IsOptional()
+  public twitterUrl?: string;
+
+  constructor(
+    tokenName: string = "",
+    tokenSymbol: string = "",
+    tokenDescription: string = "",
+    tokenImage: string = "",
+    preBuyAmount: BigNumber = new BigNumber(0)
+  ) {
+    super();
+    this.tokenName = tokenName;
+    this.tokenSymbol = tokenSymbol;
+    this.tokenDescription = tokenDescription;
+    this.tokenImage = tokenImage;
+    this.preBuyAmount = preBuyAmount;
+  }
+}
+
+export class CreateSaleResponse {
+  image: string;
+  tokenName: string;
+  symbol: string;
+  description: string;
+  websiteUrl?: string;
+  telegramUrl?: string;
+  twitterUrl?: string;
+  initialBuyAmount: string;
+  vaultAddress: string;
+  creatorAddress: string;
+}
+
+export class ExactTokenAmountDto extends ChainCallDTO {
+  @IsString()
+  @IsNotEmpty()
+  public vaultAddress: string;
+
+  @BigNumberProperty()
+  @IsNonZeroBigNumber({ message: "tokenAmount cannot be zero" })
+  @IsNotEmpty()
+  public tokenAmount: BigNumber;
+
+  @BigNumberProperty()
+  @IsOptional()
+  public expectedNativeToken?: BigNumber;
+
+  constructor(vaultAddress: string = "", tokenAmount: BigNumber = new BigNumber(0)) {
+    super();
+    this.vaultAddress = vaultAddress;
+    this.tokenAmount = tokenAmount;
+  }
+}
+
+export class NativeTokenAmountDto extends ChainCallDTO {
+  @IsString()
+  @IsNotEmpty()
+  public vaultAddress: string;
+
+  @BigNumberProperty()
+  @IsNonZeroBigNumber({ message: "nativeTokenAmount cannot be zero" })
+  @IsNotEmpty()
+  public nativeTokenAmount: BigNumber;
+
+  @BigNumberProperty()
+  @IsOptional()
+  public expectedToken?: BigNumber;
+
+  constructor(vaultAddress: string = "", nativeTokenAmount: BigNumber = new BigNumber(0)) {
+    super();
+    this.vaultAddress = vaultAddress;
+    this.nativeTokenAmount = nativeTokenAmount;
+  }
+}
+
+export class TradeResponse {
+  public tokenBalance?: TokenBalance;
+  public nativeTokenBalance?: TokenBalance;
+
+  public tokenAmount?: string;
+  public nativeTokenAmount?: string;
```

Comment:
```suggestion
  public nativeTokenQuantity?: string;
```

Within this SDK the semantic `quantity` is preferred over `amount`. 

See `chain-api/src/types/TokenBalance.ts` as an example.

Please align conventions within your code accordingly.